### PR TITLE
IR-104: Use library-go in image-registry

### PR DESCRIPTION
**Pull request by Nikita757**
Replaced some of the origin-common code with more supported library-go package.

### CODE REVIEW

**Comment by dmage** [comment]
@Nikita757 this PR doesn't delete any function/package from this repository, can't we delete something?

**Comment by dmage** [comment]
@Nikita757 please update README files for kubernetes-common and origin-common, and rebase it on top of master so that you don't have a merge commit in the PR.

File: pkg/dockerregistry/server/quotarestrictedblobstore.go [hunk 1]
```diff
@@ -153,7 +153,8 @@ func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
 
 // admitImage checks if the size is greater than the limit range.
 func admitImage(size int64, limit corev1.LimitRangeItem) error {
-	if limit.Type != imageapi.LimitTypeImage {
+	if limit.Type !=
+		imageapiv1.LimitTypeImage {
```
Comment (dmage) [inline pkg/dockerregistry/server/quotarestrictedblobstore.go:157]: Why on a different line?

File: pkg/dockerregistry/server/prune/prune.go [hunk 2]
```diff
@@ -229,7 +229,7 @@ func Prune(ctx context.Context, registry distribution.Namespace, registryClient
 			return fmt.Errorf("failed to parse the repo name %s: %v", repoName, err)
 		}
 
-		ref, err := imageapi.ParseDockerImageReference(repoName)
+		ref, err := imageref.Parse(repoName)
```
Comment (dmage) [inline pkg/dockerregistry/server/prune/prune.go:232]: Should it be imageutil.ParseDockerImageReference?

File: pkg/dockerregistry/server/quotarestrictedblobstore.go (hunk 1 above)
Comment (Nikita757) [reply pkg/dockerregistry/server/quotarestrictedblobstore.go]: A typo

File: pkg/dockerregistry/server/prune/prune.go (hunk 2 above)
Comment (Nikita757) [reply pkg/dockerregistry/server/prune/prune.go:232]: Indeed imageref.Parse() is a better choice. I'll replace them.

**Comment by xiuwang** [comment]
Do regression for image-registry, no issue found.
/lgtm

**Pull request merged**